plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -163,11 +178,11 @@ void startDevServer(List<File> runnables, MessageConsoleStream console, String h
    *
    * @param runnables the path to directories that contain configuration files like appengine-web.xml
    * @param console the stream (Eclipse console) to send development server process output to
-   * @param host the host name to which application modules should bind
+   * @param port the app service port
```

Comment:
Why are we determining the port in the launch configuration instead of here in the server implementation? Especially since we're using a server attribute to fetch the port?

(We should be able to start the server without the launch configuration.)

We should have some methods to return the actual ports used. That would simplify launch code where you're passing around the activePort.